docs/content/topics/operators/gateways.md

```diff
@@ -11,340 +11,518 @@ aliases:
+  if err != nil {
+    return err
+  }
+  
+  // If the returned events list has no items, no event was created
+  if len(events.Items) != 1 {
+    fmt.Printf(
+      "No event was created. "+
+        "Does project %s subscribe to events of source %s and type %s?\n",
+      projectID,
+      event.Source,
+      event.Type,
+    )
+    return nil
+  }
+
+  // The Brigade event was successfully created!
+  fmt.Printf(
+    "Event created with ID %s for project %s\n",
+    events.Items[0].ID,
+    projectID,
+  )
+  return nil
+}
+```
 
-```Dockerfile
-FROM debian:jessie-slim
-RUN apt-get update && apt-get install -y uuid-runtime curl
-RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl \
-  && mv kubectl /usr/local/bin/kubectl && chmod 755 /usr/local/bin/kubectl
-COPY ./cron-event.sh /usr/local/bin/cron-event.sh
-CMD /usr/local/bin/cron-event.sh
+Let's briefly look at the Brigade Event object from above.
+
+```go
+  // Construct a Brigade Event
+  event := core.Event{
+    // This is the ID/name of the project that the event will be intended for
+    ProjectID: projectID,
+    // This is the source value for this event
+    Source:    "brigade.sh/example-gateway",
+    // This is the event's type
+    Type:      "create-event",
+    // This is the event's payload
+    Payload:   "Dolly",
+  }
 ```
 
-(The really long line just installs `kubectl`)
+We've filled in the core fields needed for any Brigade event, `ProjectID`,
+`Source` and `Type`. As a bonus, we're also adding a `Payload`. That's just the
+start of what a Brigade Event can contain, however. Other notable fields worth
+researching are:
 
-And we can pack that into a Docker image by running `docker build -t technosophos/example-cron:latest .`.
-You should replace `technosophos` with your Dockerhub username (or modify the
-above to store in your Docker registry of choice).
+- `Qualifiers`: A list of qualifier values. For a project to receive events
+  from a gateway, the qualifiers on an event must exactly match the qualifiers
+  set on the project's event subscription, when provided.
 
-Then push the image to a repository that your Kubernetes cluster can reach:
+- `Labels`: A list of labels. Projects can choose to utilize these for
+  filtering purposes. In contrast to qualifiers, project event subscription
+  labels don't need to exactly match event labels in order to receive them, as
+  long as the event does have any/all labels that the project might use for
+  filtering.
 
+- `ShortTitle`: A short title for the event.
+
+- `LongTitle`: A longer, more descriptive title for the event.
+
+- `SourceState`: A key/value map representing event state that can be persisted
+  by the Brigade API server so that gateways can track event handling progress
+  and perform other actions, such as updating upstream services.
+
+### Linking it all together
```

Comment:
I think if we source the API token from and env var as mentioned in another comment, and also perhaps omit projectID, this program gets even smaller...

I'm wondering... if we get it small enough, can we present it all in one shot instead of building part of it and then revisiting it?

Reply:
Yuuup, consolidated into one section now.  Thanks for the simplifications.